### PR TITLE
Updated view component not found error message

### PR DESCRIPTION
**Pull request by juunas11**
Updated error message for when a view component is not found
- The message now defines the requirements for a view component to be discovered to help debugging

Addresses #5380

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/Resources.resx [hunk 1]
```diff
@@ -131,7 +131,7 @@
     <value>Method '{0}' of view component '{1}' should be declared to return a value.</value>
   </data>
   <data name="ViewComponent_CannotFindComponent" xml:space="preserve">
-    <value>A view component named '{0}' could not be found.</value>
+    <value>A view component named '{0}' could not be found. View components must be a public concrete non-generic class, and should be decorated with ViewComponentAttribute or the class name should end in 'ViewComponent'. The class must not be decorated with NonViewComponentAttribute.</value>
```
Comment (kichalla) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/Resources.resx]: nit: remove comma before 'and'

**Comment by kichalla** [comment]
Looks good to me. @rynowak @NTaylorMullen

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/Resources.resx (hunk 1 above)
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/Resources.resx]: `View components should be a public concrete non-generic class, and should be decorated with {1} or have a class name ending in '{2}'. The class cannot be decorated with {3}.`

**Comment by tmasyam** [comment]
check in Application logic

**Comment by jbagga** [comment]
Thank you for your input. We changed the error message to a variation of the one proposed here. See https://github.com/aspnet/Mvc/pull/6263 for details

**Pull request closed without merging**